pkgs/tools/admin/certbot/default.nix

```diff
@@ -38,6 +40,25 @@ buildPythonApplication rec {
 
   checkInputs = [ pytest_xdist pytest dateutil ];
 
+  # certbot.withPlugins has a similar calling convention as python*.withPackages
+  # it gets invoked with a lambda, and invokes that lambda with the python package set matching certbot's:
+  # certbot.withPlugins (cp: [ cp.certbot-dns-foo ])
+  passthru.withPlugins = f: let
```

Comment:
Why not call `python.withPackages` with your plugins and symlink the entry point of interest? `withPackages` exists exactly so we get rid of this `PYTHONPATH` mangling everywhere.

Reply:
But `python.withPackages` also provides a `$out/bin/python`, so one would need to symlink /only/ the entrypoint, leading to two "intermediate" derivations, right?

Reply:
You would have `python packages -> withPackages env -> certbot entry point symlink`

Add `certbot` to the Python packages set and you use it directly from `withPackages`, no need for any additional symlinks, so `python packages -> withPackages env`

Reply:
For some reason, `certbot` does not end up in a `python.withPackages ps: [ ps.certbot ];`.

I pushed a new commit on top of the existing ones as a repro.

Reply:
it needs to use `buildPythonPackage` in order to be included

Reply:
Hmm, that's somewhat unintuitive:

> The `buildPythonApplication` function is practically the same as
`buildPythonPackage`. The main purpose of this function is to build a Python
package where one is interested only in the executables, and not importable
modules.

That's exactly my usecase. Still, I now need to use `buildPythonPackage`, just because I want to make some more `PYTHONPATH`'s available to the entrypoint? And then I still need to wrap this to an intermediate, because I don't want to expose the `python` binary etc.

That sounds more like a workaround than a preferred solution. What about something like

```
(certbot.overridePythonAttrs ( old: {
  propagatedBuildInputs = old.propagatedBuildInputs or [] ++ plugins;
}))
```

… and exposing something like this in certbot's passthru function?

Reply:
That's another option, but means a rebuild instead of composition.

Reply:
Yeah, maybe we want something like `pythonApplicationWithPackages` in the longer run.

Rebuilding it for the time being seems better than changing all python applications  to use `buildPythonPackage` IMHO. I'll update the PR.

Reply:
So I tried to build this using the above

```
(certbot.overridePythonAttrs ( old: {
  propagatedBuildInputs = old.propagatedBuildInputs or [] ++ plugins;
}))
```

but the output doesn't contain binaries nor packages.

I might just revert to the initial proposal of manually adding to `PYTHONPATH`, until we have tooling for this.